app/frontend/helpers/ui_helper.js

```diff
@@ -216,6 +216,66 @@ class UiHelper {
     var textLoadingIndicator = this.getCurrentTextLoadingIndicator(tabIndex);
     textLoadingIndicator.hide();
   }
+
+  getFirstVisibleVerseAnchor() {
+    var firstVisibleVerseAnchor = null;
+    var verseListFrame = app_controller.getCurrentVerseListFrame();
+
+    if (verseListFrame != null) {
+      var verseListFrameRect = verseListFrame[0].getBoundingClientRect();
+
+      var currentNavigationPane = app_controller.navigation_pane.getCurrentNavigationPane()[0];
+      var currentNavigationPaneWidth = currentNavigationPane.offsetWidth;
+
+      var firstElementOffsetX = verseListFrameRect.x + currentNavigationPaneWidth + 15;
+      var firstElementOffsetY = verseListFrameRect.y + 15;
```

Comment:
Can use please add a comment for the "magic number". Maybe these numbers are more relevant in the next line, i.e.
```
const firstElement = document.elementFromPoint(firstElementOffsetX + 15, firstElementOffsetY + 15);
```

Reply:
Done